examples/tutorials/workgraphs/descriptors_filter_qe.ipynb

```diff
@@ -0,0 +1,533 @@
+    "from aiida.orm import Str, Float, Bool, Int\n",
+    "\n",
+    "inputs = {\n",
+    "    \"code\": janus_code,\n",
+    "    \"model\": model,\n",
+    "    \"arch\": Str(model.architecture),\n",
+    "    \"device\": Str(\"cpu\"),\n",
+    "    \"fmax\": Float(0.1), \n",
+    "    \"opt_cell_lengths\": Bool(False), \n",
+    "    \"opt_cell_fully\": Bool(True), \n",
+    "    \"metadata\": {\"options\": {\"resources\": {\"num_machines\": 1}}},\n",
+    "    }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1d661593",
+   "metadata": {},
+   "source": [
+    "We must now choose the calculations to perform:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0e4c78c9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from aiida.plugins import CalculationFactory\n",
+    "geomoptCalc = CalculationFactory(\"mlip.opt\")\n",
+    "descriptorsCalc = CalculationFactory(\"mlip.descriptors\")\n",
+    "trainCalc = CalculationFactory(\"mlip.train\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "37df1f81",
+   "metadata": {},
+   "source": [
+    "Setup graph task, which sets up QE params and created a PwCalc for each structure in file. Returns test_file, train_file and valid_file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1aff44c4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from aiida_workgraph import task\n",
+    "from aiida_workgraph.manager import get_current_graph\n",
+    "from aiida.orm import StructureData, load_group, KpointsData, SinglefileData\n",
+    "from ase.io import iread\n",
+    "from pathlib import Path\n",
+    "import yaml\n",
+    "from aiida_quantumespresso.calculations.pw import PwCalculation\n",
+    "from sample_split import process_and_split_data\n",
+    "\n",
+    "\n",
+    "@task.graph(outputs = [\"test_file\", \"train_file\", \"valid_file\"])\n",
+    "def qe(**inputs):\n",
+    "\n",
+    "    qe_code = load_code(\"qe@scarf\")\n",
+    "    wg = get_current_graph()\n",
+    "\n",
+    "    kpoints = KpointsData()\n",
+    "    kpoints.set_kpoints_mesh([1, 1, 1])\n",
+    "\n",
+    "    metadata =  {\n",
+    "        \"options\": {\n",
+    "            \"resources\": {\n",
+    "                \"num_machines\": 1,\n",
+    "                'num_mpiprocs_per_machine': 32,\n",
+    "            },\n",
+    "            'max_wallclock_seconds': 3600,         # Set maximum wallclock time\n",
+    "            'queue_name': 'scarf',                  # Set queue name\n",
+    "            'qos': 'scarf',\n",
+    "            'environment_variables': {},\n",
+    "            'withmpi': True,                       # Use MPI\n",
+    "            'prepend_text': '''\n",
+    "            module purge\n",
+    "            module use /work4/scd/scarf562/eb-common/modules/all\n",
+    "            module load amd-modules\n",
+    "            module load QuantumESPRESSO/7.2-foss-2023a\n",
```

Comment:
I thought things like this were already set up when we set the `prepend_text` when we set up the remote computer?

As a related point, I just realised it looks like we set that for the computer, rather than the code itself. Why?